pkgs/development/libraries/lightgbm/default.nix

```diff
@@ -0,0 +1,142 @@
+    ++ lib.optionals openclSupport [ opencl-headers ocl-icd boost ]
+    ++ lib.optionals mpiSupport [ openmpi ]
+    ++ lib.optionals hdfsSupport [ hadoop ]
+    ++ lib.optionals (hdfsSupport || javaWrapper) [ openjdk ]
+    ++ lib.optionals javaWrapper [ swig ]
+    ++ lib.optionals rLibrary [ R ];
+
+  buildInputs = [ gtest ]
+    ++ lib.optional cudaSupport cudaPackages.cudatoolkit;
+
+  propagatedBuildInputs = lib.optionals rLibrary [
+    rPackages.data_table
+    rPackages.jsonlite
+    rPackages.Matrix
+    rPackages.R6
+  ];
+
+  # Skip APPLE in favor of linux build for .so files
+  postPatch = ''
+    export PROJECT_SOURCE_DIR=./
+    substituteInPlace CMakeLists.txt \
+      --replace "find_package(GTest CONFIG)" "find_package(GTest REQUIRED)" \
+      --replace "OpenCL_INCLUDE_DIRS}" "OpenCL_INCLUDE_DIRS}" \
+      --replace "elseif(APPLE)" "elseif(APPLESKIP)"
+    substituteInPlace \
+      external_libs/compute/include/boost/compute/cl.hpp \
+      external_libs/compute/include/boost/compute/cl_ext.hpp \
+      --replace "include <OpenCL/" "include <CL/"
+    substituteInPlace build_r.R \
+      --replace "file.path(getwd(), \"lightgbm_r\")" "'$out/tmp'" \
+      --replace \
+        "install_args <- c(\"CMD\", \"INSTALL\", \"--no-multiarch\", \"--with-keep.source\", tarball)" \
+        "install_args <- c(\"CMD\", \"INSTALL\", \"--no-multiarch\", \"--with-keep.source\", \"-l $out/library\", tarball)"
+  '';
+
+  cmakeFlags = lib.optionals doCheck [ "-DBUILD_CPP_TEST=ON" ]
+    ++ lib.optionals cudaSupport [ "-DUSE_CUDA=1" "-DCMAKE_CXX_COMPILER=${cudaPackages.cudatoolkit.cc}/bin/cc" ]
+    ++ lib.optionals openclSupport [ "-DUSE_GPU=ON" ]
+    ++ lib.optionals mpiSupport [ "-DUSE_MPI=ON" ]
+    ++ lib.optionals hdfsSupport [
+      "-DUSE_HDFS=ON"
+      "-DHDFS_LIB=${hadoop}/lib/hadoop-3.3.1/lib/native/libhdfs.so"
+      "-DHDFS_INCLUDE_DIR=${hadoop}/lib/hadoop-3.3.1/include" ]
+    ++ lib.optionals javaWrapper [ "-DUSE_SWIG=ON" ]
+    ++ lib.optionals rLibrary [ "-D__BUILD_FOR_R=ON" ];
+
+  configurePhase = lib.optionals rLibrary ''
+    export R_LIBS_SITE="$out/library:$R_LIBS_SITE''${R_LIBS_SITE:+:}"
+  '';
+
+  # set the R package buildPhase to null because lightgbm has a
+  # custom builder script that builds and installs in one step
+  buildPhase = lib.optionals rLibrary ''
+  '';
+
+  inherit doCheck;
+
+  installPhase = ''
+      runHook preInstall
+    '' + lib.optionalString (!rLibrary) ''
+      mkdir -p $out
+      mkdir -p $out/lib
+      mkdir -p $out/bin
+      cp -r ../include $out
+      install -Dm755 ../lib_lightgbm.so $out/lib/lib_lightgbm.so
+      install -Dm755 ../lightgbm $out/bin/lightgbm
+    '' + lib.optionalString javaWrapper ''
+      cp -r java $out
+      cp -r com $out
+      cp -r lightgbmlib.jar $out
+    '' + ''
+    '' + lib.optionalString javaWrapper ''
+      cp -r java $out
+      cp -r com $out
+      cp -r lightgbmlib.jar $out
+    '' + lib.optionals rLibrary ''
```

Comment:
```suggestion
    '' + lib.optionalString rLibrary ''
```

Reply:
Thanks for the help! Giving this a try now.